src/core/main.c

```diff
@@ -952,12 +963,9 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
 
                 case ARG_CONFIRM_SPAWN:
-                        r = optarg ? parse_boolean(optarg) : 1;
-                        if (r < 0) {
-                                log_error("Failed to parse confirm spawn boolean %s.", optarg);
-                                return r;
-                        }
-                        arg_confirm_spawn = r;
+                        r = parse_confirm_spawn(optarg, &arg_confirm_spawn);
```

Comment:
the old could could dea with oparg being NULL, the new code can't! please handle the NULL case here properly!